modules/tinymce/src/core/main/ts/init/Render.ts

```diff
@@ -192,6 +202,11 @@ const render = function (editor: Editor) {
     return;
   }
 
+  editor.ui = {
```

Comment:
This will definitely cause a breaking change...  as many times we've told customers to do something like this:
```
setup: (editor) => {
  editor.ui.registry.addButton('custombutton', spec);
}
```

So I'm afraid this needs to remain in the Editor constructor and you'll need to mutate the `ui` object here.

Reply:
Fixed.

Reply:
To create the Ui then mutate it, I have to start off like this:

```
  editor.ui = {
    registry: registry()
  } as any;
```

or make `styleSheetLoader` and optional property on Ui:

```
export interface Ui {
  registry: Registry.Registry;
  styleSheetLoader?: StyleSheetLoader;
}
```

Both are pretty gross and I wonder if this is worth it. I mean, all we're really doing this for is so we can say `editor.ui.styleSheetLoader`. 

Any opinion on this issue, or any alternative ideas?

which is pretty awful.

Reply:
Also consider: if `editor.ui.registry` is supposed to be used before `render()` then maybe someone might assume that `editor.ui.styleSheetLoader` is also available at that time.

Reply:
I think the options include:

1. Mutate, as above, using `any`.
2. Mutate, as above, using `?:`
3. Make `StyleSheetLoaderRegistry` public API and don't add anything on Editor. The main problem with this is that other callers need to wire the 2 editor settings through.
4. Add it somewhere else on the Editor API. 
5. Add an editor.uidom

Reply:
I've experimented with Option 3 in the current branch, and Option 4 in branch feature/TINY-6089-3

Reply:
Hmm... Spocke's idea might work, otherwise what you've done is fine for now I think. We probably need to take a look at all this at some stage in the future, but certain things being ready at different times, isn't a huge issue atm (just so long as it's all there before init).

Reply:
I'm happy with the mutation for now. See other comment - I have a spike of Spocke's idea.